fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -580,6 +594,102 @@ where
     }
 }
 
+impl<S, Tx, Ecal> Interpreter<S, Tx, Ecal>
+where
+    S: InterpreterStorage,
+{
+    fn upload_inner(
+        upload: &mut Upload,
+        storage: &mut S,
+        initial_balances: InitialBalances,
+        gas_costs: &GasCosts,
+        fee_params: &FeeParameters,
+        base_asset_id: &AssetId,
+        gas_price: Word,
+    ) -> Result<(), InterpreterError<S::DataError>> {
+        let root = *upload.bytecode_root();
+        let bytecode = storage
+            .storage_as_ref::<UploadedBytecodes>()
+            .get(&root)
+            .map_err(RuntimeError::Storage)?
+            .map(|x| x.into_owned())
+            .unwrap_or_else(|| UploadedBytecode::Uncompleted {
+                bytecode: vec![],
+                uploaded_parts_number: 0,
+            });
+
+        let new_bytecode = match bytecode {
+            UploadedBytecode::Uncompleted {
+                mut bytecode,
+                uploaded_parts_number,
+            } => {
```

Comment:
Can we put this branch into a helper function? `upload_bytecode_subsection` or something like that.